### PR TITLE
fix: hide your dimensions button when no dimensions are available

### PR DESCRIPTION
**Pull request by edoardo**
### Key features

1. hide your dimensions button if no your dimensions are available

---

### Description

Moved the menu item for Your dimensions into a separate component which queries the `dimensions` endpoint to check if any dimension is available.
If not, hide the button as clicking it would result in an empty list in the panel.

### CODE REVIEW

File: src/components/MainSidebar/YourDimensionsPanel/useYourDimensions.js [hunk 1]
```diff
@@ -2,22 +2,25 @@ import { useDataQuery } from '@dhis2/app-runtime'
 import { useEffect, useState } from 'react'
 import { DIMENSION_LIST_FIELDS } from '../DimensionsList/index.js'
 
+const yourDimensionsResource = 'dimensions'
+// Fixed filter on org units for 2.38 ?
+const yourDimensionsFilter = 'dimensionType:eq:ORGANISATION_UNIT_GROUP_SET'
```
Comment (HendrikThePendric) [inline src/components/MainSidebar/YourDimensionsPanel/useYourDimensions.js:7]: Perhaps we should use `SCREAMING_SNAKE_CASE` for these string constants, as we do elsewhere?

File: src/components/MainSidebar/YourDimensions.js [hunk 2]
```diff
@@ -0,0 +1,42 @@
...
+        params: {
+            page: 1,
+            pageSize: 1,
+            fields: 'id',
+            filter: yourDimensionsFilter,
+        },
+    },
+}
+
+export const YourDimensions = ({ selected, count, onClick }) => {
```
Comment (HendrikThePendric) [inline src/components/MainSidebar/YourDimensions.js:24]: Perhaps consider renaming to sth like `YourDimensionsMenuItem`?

**Review by HendrikThePendric — APPROVED** [review]
Left two small suggestions, but LGTM 💪 .

File: src/components/MainSidebar/YourDimensionsPanel/useYourDimensions.js (hunk 1 above)
Comment (edoardo) [reply src/components/MainSidebar/YourDimensionsPanel/useYourDimensions.js]: Of course.
Fixed.

File: src/components/MainSidebar/YourDimensions.js (hunk 2 above)
Comment (edoardo) [reply src/components/MainSidebar/YourDimensions.js]: Done.

**Pull request merged**